velox/dwio/common/CacheInputStream.cpp

```diff
@@ -312,6 +312,8 @@ bool CacheInputStream::loadFromSsd(
                         fileIds().string(fileNum_));
       // Remove the non-loadable entry so that next access goes to storage.
       file.erase(cache::RawFileCacheKey{fileNum_, region.offset});
+      pin_ = std::move(pins[0]);
```

Comment:
The data corruption error is not handled in DirectInputStream::loadPosition(), CachedBufferedInput::makeLoads(prefetch path?), and DirectBufferedInput::makeLoads? Thanks!

Reply:
Could you help me understand how DirectInputStream/DirectBufferedInput connect to cache?

Reply:
That is probably for storage read error which should catch error to prevent server crash.